server/src/main/java/org/elasticsearch/common/util/concurrent/PrioritizedEsThreadPoolExecutor.java

```diff
@@ -46,12 +45,10 @@ public PrioritizedEsThreadPoolExecutor(
         TimeUnit unit,
         ThreadFactory threadFactory,
         ThreadContext contextHolder,
-        ScheduledExecutorService timer,
-        StarvationWatcher starvationWatcher
```

Comment:
The starvation-watching mechanism is now built into the `MasterService` itself, not the executor, so this parameter is unnecessary and removed everywhere.